scenes/lessons/7th_class/a_b+c+d/a_b+c+d.py

```diff
@@ -0,0 +1,616 @@
+        form_2[9 + 4].move_to(eq_sign_pos).align_to(form_1[9 + 4], DOWN)
+        form_2[10 + 4:13 + 4].move_to(rect_abc_copy[0].get_edge_center(UP)).align_to(form_1[0], DOWN)
+        form_2[13 + 4].move_to(pl_sign_loc).align_to(form_1[13 + 4], DOWN)
+        form_2[14 + 4:17 + 4].move_to(rect_abc_copy[1].get_edge_center(UP)).align_to(form_1[0], DOWN)
+        form_2[17 + 4].move_to(pl2_sign_loc).align_to(form_1[13 + 4], DOWN)
+        form_2[18 + 4:].move_to(rect_3_copy.get_edge_center(UP)).align_to(form_1[0], DOWN)
+
+        self.play(
+            ReplacementTransform(form_1, form_2),
+            Write(VGroup(rect_3_copy, a_copy2, d_copy)),
+            Write(VGroup(rect_3, d)),
+            Write(plus_sign_2.move_to(pl2_sign_loc))
+        )
+        self.wait(0.5)
+
+        rect_new = Rectangle(Color(WHITE), 5 * SCALE, 10 * SCALE, fill_opacity=opc)
+        rect_new.set_fill(color=[ORANGE, BLUE, RED], opacity=1)
+        rect_abcd = VGroup(
+            rect_new,
+            a.copy().move_to(rect_new.get_edge_center(LEFT) + RIGHT * 0.2),
+            get_tex('b+c+d    ').scale(0.6).set_opacity(opc).move_to(rect_new.get_edge_center(DOWN)).align_to(b, DOWN)
+        ).copy().align_to(rect_abc, UL)
+
+        self.play(
+            AnimationGroup(
+                FadeIn(rect_abcd, run_time=1.5),
+                FadeOut(VGroup(rect_abc, rect_3, d), run_time=1),
+                lag_ratio=0.25
+            )
+        )
+        self.wait()
+
+        self.play(Circumscribe(form_2[0:1], fade_out=True))
+        self.wait(0.25)
+        self.play(Circumscribe(form_2[2:11], fade_out=True))
+        self.wait()
+        self.play(
+            Wiggle(rect_abc_copy[0]),
+            Wiggle(rect_abc_copy[1]),
+            Wiggle(rect_3_copy)
+        )
+        self.wait()
+        self.play(
+            Circumscribe(form_2[14:17], fade_out=True),
+            Indicate(VGroup(rect_abc_copy[0], rect_abc_copy[2], rect_abc_copy[3]), color=None)
+        )
+        self.wait(0.25)
+        self.play(
+            Circumscribe(form_2[18:21], fade_out=True),
+            Indicate(VGroup(rect_abc_copy[1], rect_abc_copy[4], rect_abc_copy[5]), color=None)
+        )
+        self.wait(0.25)
+        self.play(
+            Circumscribe(form_2[22:-1], fade_out=True),
+            Indicate(VGroup(rect_3_copy, d_copy, a_copy2), color=None)
+        )
+        self.wait()
+
+        self.play(Circumscribe(form_2[10 + 4:]))
+        self.wait()
+
+        self.play(Circumscribe(form_2[:9 + 4]))
+        self.wait()
+        self.play(Circumscribe(form_2[10 + 4:]))
+        self.wait()
+        self.play(Wiggle(form_2[9 + 4], scale_value=1.8))
+        self.wait()
+
+        shapes = VGroup(
+            rect_abcd, rect_abc_copy,
+            VGroup(rect_3_copy, a_copy2, d_copy), d_copy,
+            equal_sign_1, plus_sign_1, plus_sign_2
+        )
+
+        shapes.save_state()
```

Comment:
What does this do?

Reply:
էդ պահի դիրքը պահում ա օբյեկտիդ, հետո վրեն ինչ-որ անիմացիաներ ես անում, հետո որ restore ես անում, հետ ա բերում էս վիճակին

Reply:
:O աստվածային 2